src/ddprof_cmdline.c

```diff
@@ -36,44 +44,208 @@ bool arg_yesno(const char *str, int mode) {
   return false;
 }
 
-bool process_event(const char *str, const char **lookup, size_t sz_lookup,
-                   size_t *idx, uint64_t *value) {
+// If this returns false, then the passed watcher should be regarded as invalid
+bool watcher_from_event(const char *str, PerfWatcher *watcher) {
+  const PerfWatcher *tmp_watcher;
+  if (!(tmp_watcher = ewatcher_from_str(str)))
+    return false;
+
+  // Now we have to process options out of the string
+  char *str_chk;                    // for checking the result of parsing
+  char *str_tmp = strchr(str, ','); // points to ',' or is nullptr
+  uint64_t value_tmp = tmp_watcher->sample_period; // get default val
+
+  if (str_tmp) {
+    ++str_tmp; // nav to after ','
+    value_tmp = strtoll(str_tmp, &str_chk, 10);
+    if (*str_chk)
+      return false; // If this is malformed, the whole thing is malformed?
+  }
+
+  // If we're here, then we've processed the event specification correctly, so
+  // we copy the tmp_watcher into the storage given by the user and update the
+  // mutable field
+  *watcher = *tmp_watcher;
+  watcher->sample_period = value_tmp;
+
+  // If an event doesn't have a well-defined profile type, then it gets
+  // registered as a tracepoint profile.  Make sure it has a valid name for the
+  // label
+  static const char event_groupname[] = "custom_events";
+  if (watcher->profile_id == DDPROF_PWT_TRACEPOINT) {
+    watcher->tracepoint_name = watcher->desc;
+    watcher->tracepoint_group = event_groupname;
+  }
+  return true;
+}
+
+#ifdef __x86_64__
+int arg2reg[] = {-1, 5, 4, 3, 2, 16, 17}; // 1-indexed, so pad
```

Comment:
Are these actual register numbers or the one that we capture through perf events ? I did a commit that might have been wrong. Could you check the following proposal ?
https://github.com/DataDog/ddprof/commit/a3c780d83b262a4880d6ee43c6bd054f886d4df6

Reply:
Well, the -1 is a placeholder to make things fail.  The next registers are numbered positionally in the lookup (they appear to correspond to the linked commit).  The last two, however, are wrong; I counted them incorrectly--good catch!

Reply:
Why not use enum values from `perf_archmap.h` ?

Reply:
Agree, fixed